Android/notification-hubs-sdk-e2etestapp/src/main/java/com/microsoft/windowsazure/messaging/e2etestapp/ApplicationContext.java

```diff
@@ -81,4 +84,12 @@ public static String getLogPostURL() {
 	public static String getGCMSenderId() {
 		return PreferenceManager.getDefaultSharedPreferences(mContext).getString(Constants.PREFERENCE_GCM_SENDER_ID, "");
 	}
+
+	public static String getFCMSenderId() {
+		return PreferenceManager.getDefaultSharedPreferences(mContext).getString(Constants.PREFERENCE_FCM_SENDER_ID, "");
+	}
+
+	public static Boolean getUseGcm(){
+		return PreferenceManager.getDefaultSharedPreferences(mContext).getBoolean(Constants.PREFERENCE_USE_DEPRECATED_GCM, false);
```

Comment:
~I guess storing the type is better than just bool~
Let's just check if the id exists

Reply:
SwitchPreference saves true/false if we will not do additional handling - in that case we need in getBoolean. When value is missing the default value "false" is passed as second argument to function getBoolean(..., false).